modules/noti/slack_client.py

```
@@ -0,0 +1,78 @@
+
+    if _cooldown_active(redis_client, cooldown_key, cooldown_sec):
+        logger.info(f"slack notify skipped (cooldown: {cooldown_key})")
+        return False
+
+    payload = json.dumps({"text": text}).encode("utf-8")
+    req = urlrequest.Request(
+        webhook,
+        data=payload,
+        headers={"Content-Type": "application/json"},
+        method="POST",
+    )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)